internal/pkg/telemetry/disable.go

```diff
@@ -2,23 +2,27 @@ package telemetry
 
 import (
 	"context"
+	"go.opentelemetry.io/otel/trace/embedded"
+	"go.opentelemetry.io/otel/trace/noop"
 
 	"go.opentelemetry.io/otel/trace"
 )
 
 const disabledTracingCtxKey = ctxKey("disabled-tracing")
 
 //nolint:gochecknoglobals
-var nopTracer = trace.NewNoopTracerProvider().Tracer("")
+var nopTracer = noop.NewTracerProvider().Tracer("")
 
 // contextTracer wraps trace.Tracer and adds the option to turn off tracing based on the context, see IsTracingDisabled function.
 // In the OpenTelemetry it is not implemented for Go yet: https://github.com/open-telemetry/opentelemetry-specification/issues/530
 type contextTracer struct {
+	embedded.Tracer
 	wrapped trace.Tracer
 }
```

Comment:
Minor changes in OTEL. They added some marker interfaces and moved `nop` tracer.